model_benchmarking/preprocessing_scripts/label_quailty_check.py

```diff
@@ -0,0 +1,438 @@
+    else:
+        cohen_kappa = cohen_kappa_score(old_count, new_count, weights = "linear")
+        icc = intraclass_corr(old_count, new_count)
+    
+    total_iou = 0
+    n_labels = 0
+    for i in range(len(old_labels)):
+        if len(old_labels[i]) == 0 and len(new_labels[i]) == 0:
+            continue
+        elif len(old_labels[i]) == 0 or len(new_labels[i]) == 0:
+            total_iou += 0
+            n_labels += 1
+        elif len(old_labels[i]) == 1 and len(new_labels[i]) == 1:
+            curr_iou = iou_for_bboxes(old_labels[i][0], new_labels[i][0])
+            if curr_iou > 0:
+                total_iou += curr_iou
+            else:
+                total_iou += 0
+            n_labels += 1
+        else:
+            old_label = sorted(old_labels[i], key=lambda x: x[0])
+            new_label = sorted(new_labels[i], key=lambda x: x[0])
+            curr_iou = iou_for_multilabels_on_one_image(old_label, new_label)
+            if curr_iou < 0:
+                total_iou += 0
+            else:
+                total_iou += curr_iou
+            n_labels += 1
+                
+    if n_labels > 0:
+        iou = total_iou / n_labels
+    else:
+        iou = "N/A"
+    
+    return num_images, mae, mse, accuracy, f1, cohen_kappa, icc, iou
+
+def quality_check_for_all_samples(target_dir):
+    img_lengths, all_mae, all_mse, all_accuracy, all_f1, all_cohen, all_icc, all_iou = [], [], [], [], [], [], [], []
+    for directory in os.scandir(target_dir):
+        if directory.is_dir():
+            directory_path = directory.path + "/"
+            print("\nCheck labels from directory: ", directory_path)
+            num_images, mae, mse, accuracy, f1, cohen_kappa, icc, iou = quality_check_for_folder(directory_path)
+            print("Quality check results:")
+            print("MAE:", mae, "    MSE:", mse, "   Accuracy:", accuracy, "    F1-score:", f1)
+            #print("Matthews Correlation Coefficient:", matthews_coef)
+            print("Weighted Cohen's Kappa:", cohen_kappa)
+            print("Intraclass correlation coefficient: ", icc)
+            #print("Threat Score:", csi)
+            print("IoU:", iou)
+            img_lengths.append(num_images)
+            all_mae.append(mae)
+            all_mse.append(mse)
+            all_accuracy.append(accuracy)
+            all_f1.append(f1)
+            all_cohen.append(cohen_kappa)
+            all_icc.append(icc)
+            all_iou.append(iou)
+    
+    avg_mae, avg_mse, avg_accuracy, avg_f1, avg_cohen, avg_icc, avg_iou, weight, weight_i = 0, 0, 0, 0, 0, 0, 0, 0, 0
+    for i in range(len(img_lengths)):
+        avg_mae += all_mae[i]*img_lengths[i]
+        avg_mse += all_mse[i]*img_lengths[i]
+        avg_accuracy += all_accuracy[i]*img_lengths[i]
+        avg_f1 += all_f1[i]*img_lengths[i]
+        avg_cohen += all_cohen[i]*img_lengths[i]
+        avg_icc += all_icc[i]*img_lengths[i]
+        weight += img_lengths[i]
+        if type(all_iou[i]) != str:
```

Comment:
It will return N/A when there are empty annotations in the JSON file. The case where all sampled images in a folder have no bounding boxes has been mentioned previously.